### PR TITLE
Add option to allow analyze to detect multiple faces

### PR DESCRIPTION
**Pull request by RoelofRuis**
Dear Sefik,

This PR adds the optional flag `enable_multiple` to `DeepFace.analyze`, which analyzes multiple faces in the given input image and returns a result for each one of them. If it is set to `False`, all output should remain the same as it was before (except that I renamed the keys from `instance_n` to `image_n`.

It's quite a lengthy PR, especially because I have reformatted the `functions.py` file, as it contained tabs instead of spaces.

I tried to create small commits that contain individual changes and refactors as I went along, so I would suggest to look at the individual commits + messages for the best understanding of my changes.

Furthermore I noticed that some people requested this feature before, but you mentioned it was 'out of scope' of this library, so I understand if you do not want to merge this (in that case i'll continue on my own fork) but I would like to at least share my changes with you in case you are interested.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -11,4 +11,5 @@ Flask>=1.1.2
 mtcnn>=0.1.0
 lightgbm>=2.3.1
 dlib>=19.20.0
-retina-face>=0.0.1
\ No newline at end of file
+retina-face>=0.0.1
+matplotlib==3.4.3
```
Comment (RoelofRuis) [inline requirements.txt:15]: I've added the latest matplotlib as it can come in handy for debugging. It might later be moved to 'test only' dependencies.

**Comment by serengil** [comment]
Sorry for the delay. It seems that I can spend time on 2nd week of Oct.

**Comment by RoelofRuis** [comment]
> Sorry for the delay. It seems that I can spend time on 2nd week of Oct.

No problem, thanks for the heads-up! In the mean time I'll resolve the conflicts with master.

**Comment by serengil** [comment]
Thank you for your understanding, I am going to be on a vacation for a week.

**Comment by yiouyou** [comment]
Any updates about this issue? The added feature - detect multiple faces - is quite useful!

**Comment by yiouyou** [comment]
@RoelofRuis Do you mind share your multiple faces fork with me? I'd like to try it, thanks!

**Comment by RoelofRuis** [comment]
> @RoelofRuis Do you mind share your multiple faces fork with me? I'd like to try it, thanks!

@yiouyou I think you can just refork my fork that holds this feature at https://github.com/PBWebMedia/deepface/tree/detect-multi-face

On the `develop` branch of that fork, there are some more features/changes, they are incrementally built on top of this one. You can check them out as well if you like, just take a look at the commit messages to see what I've changed/added.

**Comment by RoelofRuis** [comment]
Dear Sefik,

Let me first say I understand you having a chaotic time, so no need to feel sorry for not handling this earlier :)

But as of coming wednesday december the first, I will not have access anymore to the PBWebMedia organization and thus their fork of this project. Therefore I will close this MR.

If you have any interest in this feature in the future, please contact me and I'll see if I can invest time to resolve the current conflicts with master using a fork from my personal account and open a new MR.

**Comment by youngjae-avikus** [comment]
@RoelofRuis 

There is no multi-support for DeepFace.find(), is it only analyze?

**Pull request closed without merging**